node/src/models/Mentor.ts

```diff
@@ -81,71 +53,104 @@ export interface MentorModel extends Model<Mentor> {
+    {
+      _id: { $in: mentor.subjects },
+    },
+    null,
+    { sort: { name: 1 } }
+  );
 };
 
+// Return topics for all subjects or for one subject
+//  - one subject: sorted in subject order
+//  - all subjects: sorted alphabetically
 MentorSchema.statics.getTopics = async function (
   m: string | Mentor,
   subjectId?: string
 ): Promise<Topic[]> {
-  const mentor: Mentor =
-    typeof m === 'string' ? await this.findOne({ _id: m }) : m;
+  const mentor: Mentor = typeof m === 'string' ? await this.findById(m) : m;
+  if (!mentor) {
+    throw new Error(`mentor ${m} not found`);
+  }
   const topics: Topic[] = [];
   if (subjectId) {
     if (mentor.subjects.includes(subjectId)) {
-      topics.push(...(await SubjectModel.getTopics(subjectId)));
+      const subject = await SubjectModel.findById(subjectId);
+      topics.push(...subject.topics);
     }
   } else {
-    const subjects: Subject[] = await SubjectModel.find({
-      _id: { $in: mentor.subjects },
-    });
+    const subjects: Subject[] = await this.getSubjects(mentor);
     for (const s of subjects) {
-      topics.push(...(await SubjectModel.getTopics(s)));
+      topics.push(...s.topics);
     }
     topics.sort((a, b) => {
       return a.name.localeCompare(b.name);
     });
   }
-  const topicsIds = [...new Set(topics.map((t) => `${t._id}`))];
-  return topicsIds.map((tId) => topics.find((t) => `${t._id}` === tId));
+  const topicsIds = [...new Set(topics.map((t) => `${t.id}`))];
+  return topicsIds.map((tId) => topics.find((t) => `${t.id}` === tId));
 };
 
 MentorSchema.statics.getQuestions = async function (
   m: string | Mentor,
   subjectId?: string,
   topicId?: string,
   type?: QuestionType
-): Promise<Question[]> {
-  const mentor: Mentor =
-    typeof m === 'string' ? await this.findOne({ _id: m }) : m;
-  let questions: Question[] = [];
+): Promise<SubjectQuestion[]> {
+  const mentor: Mentor = typeof m === 'string' ? await this.findById(m) : m;
+  if (!mentor) {
+    throw new Error(`mentor ${m} not found`);
+  }
+  let sQuestions = [];
   if (subjectId) {
     if (mentor.subjects.includes(subjectId)) {
-      questions.push(...(await SubjectModel.getQuestions(subjectId, topicId)));
+      sQuestions.push(
+        ...(await SubjectModel.getQuestions(subjectId, topicId, mentor._id))
+      );
     }
   } else {
-    const subjects: Subject[] = await SubjectModel.find(
-      {
-        _id: { $in: mentor.subjects },
-      },
-      null,
-      { sort: { name: 1 } }
-    );
-    for (const s of subjects) {
-      questions.push(...(await SubjectModel.getQuestions(s, topicId)));
+    const subjects: Subject[] = await this.getSubjects(mentor);
+    for (const subject of subjects) {
+      sQuestions.push(
+        ...(await SubjectModel.getQuestions(subject, topicId, mentor._id))
+      );
     }
   }
   if (type) {
-    questions = questions.filter((q: Question) => q.type === type);
+    sQuestions = sQuestions.filter((sq) => sq.question.type === type);
```

Comment:
we should move this filter into the model/mongo query itself. Probably better to also change the argument for the `SubjectModel.getQuestions` to a single typed arg object too, e.g.

```typescript
SubjectModel.getQuestions(arg: { 
  subject: string | Subject,
  topicId?: string,
  mentorId?: string,
  type?: QuestionType
})
```